platform/openide.explorer/src/org/openide/explorer/view/OutlineView.java

```diff
@@ -269,24 +269,28 @@ public void keyReleased(KeyEvent e) {
             defaultTreeActionListener, KeyStroke.getKeyStroke(KeyEvent.VK_ENTER, 0, false), JComponent.WHEN_FOCUSED
         );
 
-        final Color focusSelectionBackground = outline.getSelectionBackground();
-        final Color focusSelectionForeground = outline.getSelectionForeground();
-        outline.addFocusListener(new java.awt.event.FocusListener(){
-            @Override
-            public void focusGained(java.awt.event.FocusEvent ev) {
-                outline.setSelectionBackground(focusSelectionBackground);
-                outline.setSelectionForeground(focusSelectionForeground);
-            }
+        // toggle selection colors depending on whether table is focused or not
+        if (!UIManager.getLookAndFeel().getID().startsWith("FlatLaf")) {
```

Comment:
The diff is a bit confusing here. Is this just skipping a bunch of code for the FlatLaf case?

Why is the code not needed for FlatLaf? Shouldn't something rather be done so that focusSelectionBackground/focusSelectionForeground ends up being the right color?

(If I remember correctly, there might very well be some bugs in OutlineView around this code; I implemented workarounds in my NetBeans Platform application many years ago. If the changes suggested here are really workarounds for bugs, then maybe the bugs should be fixed instead.)

Maybe add a copy of the isFlatLaf() method for consistency with HtmlLabelUI. That will make it easier to grep for duplicated logic in the future.

Reply:
GitHub diff is not that good. In SmartGit it looks like this:
![image](https://user-images.githubusercontent.com/5604048/72210927-b13baf80-34c2-11ea-976e-4e8691030feb.png)

> Why is the code not needed for FlatLaf?

Because FlatLaf does this itself. With the NB code enabled it did not work anymore...

But in Windows LAF it works fine. Do not think that there is a problem in the NB code.

Reply:
> Because FlatLaf does this itself. With the NB code enabled it did not work anymore...

That would be a good comment to add in the code.